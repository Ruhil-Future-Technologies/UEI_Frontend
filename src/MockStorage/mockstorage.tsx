/* eslint-disable @typescript-eslint/no-explicit-any */
import { Box } from '../Pages/AcademicHistory/AcademicHistory';

export const mockStorage: any = {
  menulist1: JSON.stringify([
    {
      created_at: 'Tue, 11 Jun 2024 12:15:27 GMT',
      form_data: {
        form_url: '/main/SuperAdmin',
        is_save: true,
        is_search: true,
        is_update: true,
      },
      id: 23,
      is_active: 1,
      is_deleted: false,
      menu_name: 'Menu',
      priority: 1,
      submenus: [
        {
          created_at: 'Mon, 24 Jun 2024 09:58:08 GMT',
          id: 56,
          is_active: 1,
          is_deleted: false,
          menu_master_id: 23,
          menu_master_name: 'User Authorization',
          menu_name: 'Menu',
          priority: 2,
          updated_at: 'Mon, 24 Jun 2024 13:37:35 GMT',
        },
      ],
      updated_at: 'Fri, 14 Jun 2024 07:40:04 GMT',
    },
    {
      created_at: 'Tue, 11 Jun 2024 12:15:27 GMT',
      form_data: {
        form_url: '/main/SuperAdmin',
        is_save: true,
        is_search: true,
        is_update: true,
      },
      id: 24,
      is_active: 1,
      is_deleted: false,
      menu_name: 'Submenu',
      priority: 1,
      submenus: [
        {
          created_at: 'Mon, 24 Jun 2024 09:58:08 GMT',
          id: 57,
          is_active: 1,
          is_deleted: false,
          menu_master_id: 24,
          menu_master_name: 'User Authorization',
          menu_name: 'Submenu',
          priority: 2,
          updated_at: 'Mon, 24 Jun 2024 13:37:35 GMT',
        },
      ],
      updated_at: 'Fri, 14 Jun 2024 07:40:04 GMT',
    },
  ]),
};

export const contextValue = {
  namepro: { id: 73, user_type: 'admin', userid: 'ashish7080@gmail.com' },
  setNamepro: jest.fn(),
  logoutpro: jest.fn(),
  setProImage: jest.fn(),
  proImage: '',
  setProPercentage: jest.fn(),
  ProPercentage: '',
  setNamecolor: jest.fn(),
  namecolor: '#ffffff',
  activeForm: 0, // Assign a number value, e.g., 0
  setActiveForm: jest.fn(), // Mock function remains unchanged
};

export const mockBoxes: Box[] = [
  {
    id: 334,
    institute_type: 'college',
    board: '',
    state_for_stateboard: '',
    institute_id: '95',
    course_id: 87,
    learning_style: 'online',
    class_id: '24',
    year: '2015-12-31T18:30:00.000Z',
    stream: '',
    university_id: '',
    sem_id: 1,
  },
];

export const mockClassValue = {
  data: [
    {
      class_name: 'class_12',
      created_at: 'Thu, 12 Dec 2024 11:43:04 GMT',
      created_by: 'afasfsafsaf',
      id: '5500819c-09e5-4543-8cf2-8f0d5cf6ffc3',
      is_active: true,
      is_deleted: false,
      updated_at: 'Thu, 12 Dec 2024 11:43:04 GMT',
      updated_by: 'Admin',
    },
  ],
  status: 200,
  message: 'Class Found Successfully',
};

export const mockbasicinfoPayload = {
  payload: [
    {
      student_login_id: '123',
      first_name: 'John',
      last_name: 'Patel',
      gender: 'male',
      dob: '01/01/2024',
      father_name: 'father',
      mother_name: 'mother',
      guardian_name: 'guardian',
      pic_path: 'pic_path',
      aim: 'aim',
    },
  ],
};
