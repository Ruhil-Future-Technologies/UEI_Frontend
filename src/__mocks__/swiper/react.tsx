import React from 'react';

export const Swiper: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => <div>{children}</div>;

export const SwiperSlide: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => <div>{children}</div>;
