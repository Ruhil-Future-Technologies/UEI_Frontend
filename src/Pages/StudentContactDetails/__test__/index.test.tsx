

import React from 'react';
import { MemoryRouter } from "react-router-dom";
import { contextValue } from "../../../MockStorage/mockstorage";
import NameContext from "../../Context/NameContext";
import { fireEvent, render, waitFor, screen } from "@testing-library/react";
import StudentcontactDetails from '..';
import useApi from '../../../hooks/useAPI';
//import { deepEqual } from '../../../utils/helpers';
jest.mock('../../../hooks/useAPI');

describe("Student Contect Component", () => {
  // Mocking the required props
  const mockSetActiveForm = jest.fn();
  const mockActiveForm = 2;
  const mockGetData = jest.fn();
  const mockPostData = jest.fn();
  const mockPutData = jest.fn();

  beforeEach(() => {
    (useApi as jest.Mock).mockImplementation(() => ({
      getData: mockGetData.mockResolvedValue({}), // Mock resolved promise
      postData: mockPostData,
      putData: mockPutData,
    }));
  });

  const renderedComponent = () => {
    return render(
      <NameContext.Provider value={contextValue}>
        <MemoryRouter>
          <StudentcontactDetails
            setActiveForm={mockSetActiveForm}
            activeForm={mockActiveForm}
          />
        </MemoryRouter>
      </NameContext.Provider>,
    );
  };

  it("Should render Student contect component correctly", () => {
    const { asFragment } = renderedComponent();
    expect(asFragment()).toMatchSnapshot();
  })

  it("should render fields Student contact correctly", () => {
    const { getByTestId, getByText, getByRole } = renderedComponent();
    const prevButten = getByRole('button', {
      name: /previous/i
    })
    const nextbutten = getByRole('button', {
      name: /next/i
    })
    expect(getByTestId("county_pcode")).toBeInTheDocument();
    expect(getByTestId("county_wpcode")).toBeInTheDocument();
    expect(getByTestId("mobile_num")).toBeInTheDocument();
    expect(getByTestId("email_id")).toBeInTheDocument();
    expect(getByTestId("whtmobile_num")).toBeInTheDocument();
    expect(getByText(/mobile number \*/i)).toBeInTheDocument();
    expect(getByText(/whatsapp number/i)).toBeInTheDocument();
    expect(getByText(/email id/i)).toBeInTheDocument();
    expect(nextbutten).toBeInTheDocument();
    expect(prevButten).toBeInTheDocument();

  })


  it("should call handleChange function on input change", async () => {
    const { getByTestId, } = renderedComponent();

    const mobileInput = getByTestId('mobile_num').querySelector('input') as HTMLInputElement;
    const whatsappNum = getByTestId('whtmobile_num').querySelector('input') as HTMLInputElement;
    const emailInput = getByTestId('email_id').querySelector('input') as HTMLInputElement;
    const countrycodeDropdown = getByTestId('county_pcode').querySelector('input') as HTMLInputElement;
    const countrycodewhatsaDropdown = getByTestId('county_wpcode').querySelector('input') as HTMLInputElement;

    fireEvent.change(mobileInput, { target: { value: '9876545678' } });
    fireEvent.change(whatsappNum, { target: { value: '6765433458' } });
    fireEvent.change(emailInput, { target: { value: 'atul@gmail.com' } });
    fireEvent.change(countrycodeDropdown, { target: { value: '+91' } });
    fireEvent.change(countrycodewhatsaDropdown, { target: { value: '+91' } });

    expect(mobileInput.value).toBe('9876545678');
    expect(whatsappNum.value).toBe('6765433458');
    expect(emailInput.value).toBe('atul@gmail.com');
    expect(countrycodeDropdown.value).toBe('+91');
    expect(countrycodewhatsaDropdown.value).toBe('+91');


  });

  it('should call SubmitHandle function when Next button is clicked', async () => {
    const mockSubmitHandle = jest.fn(); // Mock the SubmitHandle function

    const { getByTestId } = renderedComponent();

    const submitForm = getByTestId('submitForm') as HTMLButtonElement;

    submitForm.onclick = mockSubmitHandle;

    fireEvent.click(submitForm);


    await waitFor(() => {
      expect(mockSubmitHandle).toHaveBeenCalledTimes(1);
    });
  });

  it("should call setActiveForm function when previous is clicked", async () => {
    const { getByTestId } = renderedComponent();


    const button = getByTestId("gobackform");
    fireEvent.click(button);

    expect(mockSetActiveForm).toHaveBeenCalledTimes(1);

    expect(mockSetActiveForm).toHaveBeenCalledWith(expect.any(Function));


    const decrementFunction = mockSetActiveForm.mock.calls[0][0];
    expect(decrementFunction(5)).toBe(4);
  });

  it('should display validation message when mobile number is invalid', () => {
    const { getByTestId, } = renderedComponent();

    const mobileInput = getByTestId('mobile_num').querySelector('input') as HTMLInputElement;

    fireEvent.change(mobileInput, { target: { value: 'sjbvks' } });
    fireEvent.blur(mobileInput);

    expect(screen.getByText((content) => content.includes('Mobile number should be 10 digits'))).toBeInTheDocument();


    fireEvent.change(mobileInput, { target: { value: '123456789' } });
    fireEvent.blur(mobileInput);
    expect(screen.getByText((content) => content.includes('Mobile number should be 10 digits'))).toBeInTheDocument();


    fireEvent.change(mobileInput, { target: { value: '123456789012' } });
    fireEvent.blur(mobileInput);
    expect(screen.getByText((content) => content.includes('Mobile number should be 10 digits'))).toBeInTheDocument();


    fireEvent.change(mobileInput, { target: { value: '1234567890' } });
    fireEvent.blur(mobileInput);

    expect(screen.queryByText((content) => content.includes('Mobile number should be 10 digits'))).not.toBeInTheDocument();
  });
  it("should be check data comes and flag is false ", async () => {
    let flagValue = false;
    mockGetData.mockResolvedValueOnce({
      status: 200,
      data: [
        { id: 1, email_id: 'atul@gmail.com', mobile_no_call: "8975461325", mobile_isd_call: "+91", mobile_isd_watsapp: "+91", mobile_no_watsapp: "8975642563" },
      ],
    });


    const fetchData = async () => {
      const response = await mockGetData();
      if (response.status === 200 && response.data.length > 0) {
        flagValue = false;
      } else {
        flagValue = true;
      }
    };

    if (flagValue) {
      await mockPostData(); // Call Save API
    } else {
      await mockPutData(); // Call Update API
    }


    await fetchData();

    // Assertions for Update API
    expect(mockPutData).toHaveBeenCalled(); // Verify update API was called
    expect(mockPostData).not.toHaveBeenCalled()
  })
  it("should save or update data when the form is submitted", async () => {
    const mockSubmitHandle = jest.fn();
    //const mockToast = jest.fn();
  
    const { getByTestId } = renderedComponent();
  
    // Mock the initial data fetch
    mockGetData.mockResolvedValueOnce({
      status: 200,
      data: [
        { id: 1, email_id: 'atul@gmail.com', mobile_no_call: "8975461325", mobile_isd_call: "+91", mobile_isd_watsapp: "+91", mobile_no_watsapp: "8975642563" },
      ],
    });
  
    // Fetch data to determine flag value
    let flagValue = false;
    const fetchData = async () => {
      const response = await mockGetData();
      if (response.status === 200 && response.data.length > 0) {
        flagValue = true;
      } else {
        flagValue = false;
      }
    };
    await fetchData();
  
    // Mock the form submission
    const submitForm = getByTestId('submitForm') as HTMLButtonElement;
    submitForm.onclick = mockSubmitHandle;
  
    // Simulate the form submission
    await fireEvent.click(submitForm);
  
    // Mock the responses based on flagValue
    if (flagValue) {
      await mockPutData();
    } else {
      await mockPostData();;
    }
  
    // Wait for assertions
    await waitFor(() => {
      if (flagValue) {
        expect(mockPutData).toHaveBeenCalledTimes(1);
        expect(mockPostData).not.toHaveBeenCalled();
      } else {
        expect(mockPostData).toHaveBeenCalledTimes(1);
        expect(mockPutData).not.toHaveBeenCalled();
      }
  
      // Ensure that the toast notifications were triggered
      //expect(mockToast).toHaveBeenCalled();
    });
  });
  
  // it("should save or update data based on editFlag and payload comparison", async () => {
  //   const mockSubmitHandle = jest.fn();
  //   const mockToast = jest.fn();
    
  //  let  editFlag =false;
  //   // Setup initial state and payload values
  //   const initialState = {
  //     student_id: 1,
  //     mobile_isd_call: '+91',
  //     mobile_no_call: '9876543210',
  //     mobile_isd_watsapp: '+91',
  //     mobile_no_watsapp: '9876543210',
  //     email_id: 'test@example.com',
  //   };
  
  //   const payload = {
  //     student_id: 1,
  //     mobile_isd_call: '+91',
  //     mobile_no_call: '9876543210',
  //     mobile_isd_watsapp: '+91',
  //     mobile_no_watsapp: '9876543210',
  //     email_id: 'test@example.com',
  //   };
  
  //   const eq = deepEqual(initialState, payload); // This is the condition you are testing.
  
  //   const { getByTestId } = renderedComponent();
  
  //   // Mock the responses for postData and putData
    
   
   
  
  //   // Simulate the form submission
  //   const submitForm = getByTestId('submitForm') as HTMLButtonElement;
  //   submitForm.onclick = mockSubmitHandle;
  

  //   if(editFlag){
  //     mockPutData()
  //     mockSetActiveForm(3);
  //   }else{
  //     if(!eq){
  //       mockPostData();
  //     }
  //     mockSetActiveForm(3);
  //   }
  
  //   // Wait for assertions
  //   await waitFor(() => {
  //     if (editFlag) {
  //       // If editFlag is true, expect postData and toast success with save message
  //       expect(mockPostData).toHaveBeenCalledTimes(1);
  //       expect(mockPutData).not.toHaveBeenCalled();
  //       expect(mockToast).toHaveBeenCalledWith('Contact Details saved successfully');
  //     } else {
  //       if(!eq){
  //         expect(mockPostData).not.toHaveBeenCalled();
  //         expect(mockPutData).toHaveBeenCalledTimes(1);
  //         expect(mockToast).toHaveBeenCalledWith('Contact Details updated successfully');
  //       }
  //     }
  //   });
  // });
  
})