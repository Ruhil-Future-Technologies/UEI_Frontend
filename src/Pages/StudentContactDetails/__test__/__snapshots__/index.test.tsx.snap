// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Contect Component Should render Student contect component correctly 1`] = `
<DocumentFragment>
  <form>
    <div
      class="d-flex justify-content-start"
    >
      <div
        class="row"
      >
        <div
          class="form_field_wrapper"
        >
          <label
            style="text-align: left; margin: 10px;"
          >
            Mobile Number *
          </label>
        </div>
        <div
          class="col-3 form_field_wrapper"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-crb59f-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-testid="county_pcode"
            >
              <div
                aria-controls=":r0:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="combobox"
                tabindex="0"
              >
                +91
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                required=""
                tabindex="-1"
                value="+91"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="col form_field_wrapper"
        >
          <div
            class="MuiFormControl-root MuiTextField-root form-control css-1hozo93-MuiFormControl-root-MuiTextField-root"
            data-testid="mobile_num"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                name="phoneNum"
                placeholder="Enter Mobile number"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="d-flex justify-content-start"
    >
      <div
        class="row"
      >
        <div
          class="form_field_wrapper"
        >
          <label
            style="text-align: left; margin-left: 10px; margin-top: 10px; margin-bottom: 10px;"
          >
            Whatsapp Number
          </label>
        </div>
        <div
          class="col-3 form_field_wrapper"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-crb59f-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-testid="county_wpcode"
            >
              <div
                aria-controls=":r2:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="combobox"
                tabindex="0"
              >
                +91
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="+91"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="col form_field_wrapper"
        >
          <div
            class="MuiFormControl-root MuiTextField-root form-control css-1hozo93-MuiFormControl-root-MuiTextField-root"
            data-testid="whtmobile_num"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r3:"
                name="whatsappNum"
                placeholder="Enter Whatsapp number"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row d-flex justify-content-start"
    >
      <div
        class="col-lg-6 form_field_wrapper"
      >
        <label
          style="text-align: left; margin: 5px;"
        >
           Email Id 
        </label>
        <div
          class="MuiFormControl-root MuiTextField-root form-control css-1hozo93-MuiFormControl-root-MuiTextField-root"
          data-testid="email_id"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r4:"
              name="email"
              type="email"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-lg-12"
    >
      <div
        class="mt-3 d-flex align-items-center justify-content-between"
      >
        <button
          class="btn btn-outline-dark prev-btn px-lg-4  rounded-pill"
          data-testid="gobackform"
          type="button"
        >
          Previous
        </button>
        <button
          class="btn btn-dark px-lg-5  ms-auto d-block rounded-pill next-btn"
          data-testid="submitForm"
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
